timbermill-client/src/main/java/com/datorama/oss/timbermill/pipe/TimbermillClient.java

```diff
@@ -50,17 +59,24 @@ private TimbermillServerOutputPipe() {
                         sendEvents(eventsWrapper);
                     }
                 } catch (Exception e) {
-                    LOG.error("Error sending events to Timbermill server" ,e);
+                    LOG.error("Error sending events to Timbermill server", e);
                 }
             } while (keepRunning);
-        });
-        thread.setName("timbermill-sender");
-        thread.start();
+        };
+
+        for (int i = 0; i < builder.numOfThreads; i++) {
+            executorService.execute(getAndSendEventsTask);
+        }
+
         Runtime.getRuntime().addShutdownHook(new Thread(() -> {
             keepRunning = false;
+            executorService.shutdown();
             try {
```

Comment:
I noticed we don't close executorService in ElasticsearchClient,
so not sure if it's missing there or it's redundant here